### DISCUSSION TITLE
How to use References() with specifying different type?

### DISCUSSION THREAD

**Discussion by Ephasme** [nhibernate/fluent-nhibernate#427] [General] [answered]
I want to use Optional references with LanguageExt but I can't use `Option<Something> Something {get;set;}` with the mapping `References(x => x.Something).CustomType(typeof(Something))`
How could I do this?

I asked this very question on SO : https://stackoverflow.com/questions/54747437/
It's more complete than here.

**Comment by hazzik** [comment, marked as answer]
[Answered](https://stackoverflow.com/a/55600695/259946) by the @Ephasme:

> I finally found a solution:
> ```csharp
> References(x => x.Sunroof, "idSunroof")
>     .Access.CamelCaseField(Prefix.Underscore)
>     .Class<Window>();
> ```

**Comment by hazzik** [comment]
To elaborate on the answer. `CustomType` is for value types. `References` is to make a reference to an entity.
